src/ext/java/org/opentripplanner/ext/ridehailing/service/uber/UberService.java

```diff
@@ -0,0 +1,166 @@
+package org.opentripplanner.ext.ridehailing.service.uber;
+
+import static jakarta.ws.rs.core.HttpHeaders.ACCEPT_LANGUAGE;
+import static jakarta.ws.rs.core.HttpHeaders.AUTHORIZATION;
+import static jakarta.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static java.util.Map.entry;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Currency;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.opentripplanner.ext.ridehailing.CachingRideHailingService;
+import org.opentripplanner.ext.ridehailing.RideHailingServiceParameters;
+import org.opentripplanner.ext.ridehailing.model.ArrivalTime;
+import org.opentripplanner.ext.ridehailing.model.RideEstimate;
+import org.opentripplanner.ext.ridehailing.model.RideEstimateRequest;
+import org.opentripplanner.ext.ridehailing.model.RideHailingProvider;
+import org.opentripplanner.ext.ridehailing.service.oauth.OAuthService;
+import org.opentripplanner.ext.ridehailing.service.oauth.UrlEncodedOAuthService;
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+import org.opentripplanner.framework.io.HttpUtils;
+import org.opentripplanner.framework.json.ObjectMappers;
+import org.opentripplanner.transit.model.basic.Money;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of a ride hailing service for Uber.
+ */
+public class UberService extends CachingRideHailingService {
```

Comment:
This is nice, just from `UberService extends CachingRideHailingService` I kind of know what I need to do to support another service - of cause there will be some injection code as well.